src/main/java/StringManipulation.java

```diff
@@ -1,33 +1,115 @@
+extracting substrings, and restoring shuffled strings. It implements the StringManipulationInterface,
+which defines the contract for these operations.
+* */
 public class StringManipulation implements StringManipulationInterface {
 
+    String string; // holding string input
+
     @Override
-    public String getString() {
-        return null;
+    public String getString() { // method to return string
+        return string;
     }
 
     @Override
-    public void setString(String string) {
+    public void setString(String string) { // method to set the string
+        this.string = string;
     }
 
+    // this method will count the # of words in the string
     @Override
     public int count() {
-        return 0;
+        // throw null pointer exception if string is null
+        if (string == null) {
+            throw new NullPointerException("String is null");
+        }
+        // if string is empty throw an exception
+        if (string.isEmpty()) {
+            throw new IndexOutOfBoundsException("string is empty!");
+        }
+        // creating an array for the string
+        String[] words = string.split("\\s+");
+        // returning length of words(string count)
+        return words.length;
     }
 
     @Override
     public String removeNthCharacter(int n, boolean maintainSpacing) {
-        return null;
+        // if n < 0, throw an error
+        if (n <= 0) {
+            throw new IllegalArgumentException("'n' must be greater than zero.");
+        }
+        // if n > length, throw an error
+        if (n > string.length()) {
+            throw new IndexOutOfBoundsException("'n' is greater than the string length.");
+        }
+        // catching null errors
+        if (string == null) {
+            throw new NullPointerException("String is null!");
+        }
+        // string builder to manipulate
+        StringBuilder result = new StringBuilder();
+        // setting count to 1
+        int count = 1;
+        // looping through the string
+        for (int i = 0; i < string.length(); i++) {
+            if (count % n != 0) {
+                result.append(string.charAt(i));// append char to the result string
+            } else if (maintainSpacing) {
+                result.append(' ');// append a space to maintain spacing
+            }
+            count++;// increment count
+        }
+
+        return result.toString();// return modified string
     }
 
     @Override
     public String[] getSubStrings(int startWord, int endWord) {
-        return null;
+        // catching null string
+        if (string == null) {
+            throw new NullPointerException("String cannot be null!");
+        }
+        String[] words = string.trim().split("\\s+");// splitting the string into words
+        // throwing an error if out of bounds
+        if (startWord <= 0 || endWord <= 0 || startWord > endWord || endWord > words.length) {
```

Comment:
comparison can be simplified to startWord > endWord || startWord <= 0, because it only moves past the first condition if it is false, and if startWord < endWord and endWord <= 0, startWord must also be less than 0
Also, endwords > words.length comparison should not be here as it is covered in line 83 in the next if block and should throw a different exception.

Reply:
I do not see what you are talking about in line 83. There is no other exception being thrown after. I think that these operators do not need to be simplified as shown in the test case, when an end word is larger than the words in the string, it will catch the error.